service/src/test/java/uk/nhs/adaptors/gp2gp/common/storageconnectors/StorageConnectorFactoryTest.java

```diff
@@ -0,0 +1,34 @@
+package uk.nhs.adaptors.gp2gp.common.storageconnectors;
+
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.test.context.ContextConfiguration;
+
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static uk.nhs.adaptors.gp2gp.common.storageconnectors.StorageConnectorOptions.AZURE;
+import static uk.nhs.adaptors.gp2gp.common.storageconnectors.StorageConnectorOptions.S3;
+
+@ContextConfiguration
+@SpringBootTest
+public class StorageConnectorFactoryTest {
+
+    @Autowired
+    private StorageConnector storageConnector;
+    @Autowired
+    private StorageConnectorConfiguration configuration;
+
+    @Test
+    public void checkFactoryReturnsCorrectConnector() {
```

Comment:
Please, do not write your tests like that. Every single test method should perform one scenario. invoked. Mock "configuration.getPlatform" for "getObject" method for "StorageConnector". Then inject it onto object with [ReflectionTestUtils](https://www.baeldung.com/spring-reflection-test-utils). Then use [Parametrised JUnit tests](https://www.google.pl/search?q=parameterized+test+java&spell=1&sa=X&ved=2ahUKEwijzq2RxK_tAhXWTBUIHStmDpoQBSgAegQICRA1&biw=1440&bih=697) to check if every connector is returned.